src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosDriverFactory.scala

```diff
@@ -45,4 +59,33 @@ class MesosDriverFactory(val mesosScheduler: Scheduler, val frameworkInfo: Frame
     mesosDriver.get.stop(true)
     mesosDriver = None
   }
+
+
+  /**
+   * Create the optional credentials instance, used to authenticate calls from Chronos to Mesos.
+   */
+  def buildMesosCredentials(principal: String, secretFile: Option[String]): Credential = {
+
+    val credentialBuilder = Credential.newBuilder()
+      .setPrincipal(principal)
+
+    secretFile foreach { file =>
+      try {
+        val secretBytes = ByteString.readFrom(new FileInputStream(file))
+
+        val filePermissions = PosixFilePermissions.toString(Files.getPosixFilePermissions(Paths.get(file)))
+        (new Regex("^[-rwe]{6}---$")).findFirstIn(filePermissions).getOrElse(log.warning(s"Secret file $file has global permissions."))
```

Comment:
Should this be 'rwx' instead of 'rwe'?
Have you tested this with a few different permissions?
Seems like this would be more expressive if you used the PosixFilePermissions enums to check that OTHERS_READ and OTHERS_WRITE are not set.